clients/tabby-chat-panel/src/index.ts

```diff
@@ -19,24 +19,40 @@ export interface FetcherOptions {
 }
 
 export interface InitRequest {
-  message?: string
-  selectContext?: Context
-  relevantContext?: Array<Context>
-  fetcherOptions?: FetcherOptions
+  fetcherOptions: FetcherOptions
 }
 
 export interface Api {
   init: (request: InitRequest) => void
+  sendMessage: (message: ChatMessage) => void
+}
+
+export interface ChatMessage {
+  message: string
+  selectContext?: Context
+  relevantContext?: Array<Context>
 }
 
-export function createClient(endpoint: MessageEndpoint) {
-  return createEndpoint<Api>(endpoint)
+type CreateThreadFn =
+  ((target: any, opts: ThreadOptions<Api>) => Record<string, any>) |
+  ((opts: ThreadOptions<Api>) => Record<string, any>)
+
+export function createClient(createFn: CreateThreadFn, target: any) {
+  return createFn(target, {
+    callable: ['init', 'sendMessage'],
+  }) as Api
 }
```

Comment:
```suggestion
export function createClient(target: any) {
  return createThreadFromIframe(target, {
    callable: ['init', 'sendMessage'],
  }) as Api
}
```

Just use iframe. We shall use different function for vscode, e.g `createClientFrom...`